Comment: parse author ip address from endpoint responses.

ScoutHarris opened the pull request:
Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/16876
WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/16982

This parses the Comment author ip address from both REST and XMLRPC endpoint responses, and stores it in `RemoteComment`.

### Testing Details

Can be tested in referenced WPiOS PR.

ScoutHarris committed:
Parse Comment author ip address from endpoint responses.

ScoutHarris committed:
swiftlint autocorrect

dvdchr commented on WordPressKitTests/BlockEditorSettingsServiceRemoteTests.swift:
```diff
@@ -205,7 +205,7 @@ extension BlockEditorSettingsServiceRemoteTests {
         service = BlockEditorSettingsServiceRemote(remoteAPI: mockOrgRemoteApi)
 
-        service.fetchBlockEditorSettings(forSiteID: siteID) { (response) in
+        service.fetchBlockEditorSettings(forSiteID: siteID) { (_) in
```
👍

dvdchr reviewed (approved):
Tested by going through the scenarios described in https://github.com/wordpress-mobile/WordPress-iOS/pull/16982. LGTM! :shipit: 

I'm not sure about the BuildKite errors, though. Should we be concerned about this?

dvdchr committed:
Update bash-cache dependency for Buildkite

Since the branch `add/cocoapods-support` has been merged to main branch, let's update this to `v1.1.0` instead.

dvdchr committed:
Update bash-cache plugin in common plugins

dvdchr committed:
Revert bash-cache plugin changes

After discussion with @AliSoftware, this will be addressed in #426 so let's wait for it to be merged.

dvdchr commented:
@ScoutHarris, if you noticed, I made two little commits to try and unblock this PR from the Buildkite CI failures, but ended up reverting it 😅.  Buildkite jobs are non-blocking, and the issue will eventually be addressed in #426.
